wordpress-mobile/WordPress-Android
# Issue/13329 backup details viewholder

Parent #13329 

This PR adds support for the common `JetpackListItemState` and ViewHolders into the Backup Download Details area.

`CheckboxState` was added to the common states because it is used by both download & restore and it didn't require hardcoded values. I can add a "Backup" prefix like I did for the BACKUP_SUB_HEADER (which is also shared between backup & restore), but wanted to solicit opinions first. It shouldn't hold up the PR, because I can quickly follow this one with a prefix one.

Added some new packages for adapters & viewholders under jetpack -> backup -> details. I think the opportunity exists to pull these out into a common package to share between backup & restore and I will do during the restore implementation. Trying to keep consistent with packaging across all the features in jetpack.

I renamed BackupAvailableItemsProvider to JetpackAvailableItemsProvider and added it to the common -> provider package under common. It will be shared between restore and backup download. Again, this can be moved to common package under backup. I'm not married to the `JetpackAvailableItemsProvider` name or location. Wanted to get the concepts down and let the location fall about naturally. Again, would love opinions.

I added a ColorResId property to IconState for those instances in which we use a Material Design icon that needs coloring. Take a look at the implementation in `JetpackIconViewHolder`

I tried my best to keep the PR small, but just changing out ListItemUiState with the new one was large. The commits should be helpful. This PR builds upon another #13639 (details in the commit on dec 18) that sets up the VM to request data - it was prep work for this PR to work properly.

**Merge Instructions**
- Make sure #13639 has been merged
- Remove Not Ready for Merge Label
- Merge as normal

**Not addressed in this PR:**
(1) Any design work
(2) The action button is not hooked up
(3) Additional tests 

**To test:**
- Launch the app with BackupDownloadConfig flag on
- Navigate to ActivityLog
- Tap the More menu
- Tap _Download Backup_ menu item
- Note that the view is shown as expected
- Tap on a checkbox and it should ucheck

PR submission checklist:
